src/main/java/com/orbitz/consul/config/CacheConfig.java

```diff
@@ -0,0 +1,133 @@
+    }
+
+    /**
+     * Is the automatic adjustment of read timeout enabled?
+     */
+    public boolean isTimeoutAutoAdjustmentEnabled() {
+       return timeoutAutoAdjustmentEnabled;
+    }
+
+    /**
+     * Gets the margin of the read timeout for caches.
+     * The margin represents the additional amount of time given to the read timeout, in addition to the wait duration.
+     */
+    public Duration getTimeoutAutoAdjustmentMargin() {
+        return timeoutAutoAdjustmentMargin;
+    }
+
+    /**
+     * Gets the minimum time between two requests for caches.
+     */
+    public Duration getMinimumDurationBetweenRequests() {
+        return minDelayBetweenRequests;
+    }
+
+    /**
+     * Creates a new {@link CacheConfig.Builder} object.
+     *
+     * @return A new Consul builder.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Duration backOffDelay;
+        private Duration minDelayBetweenRequests;
+        private Duration timeoutAutoAdjustmentMargin;
+        private Boolean timeoutAutoAdjustmentEnabled;
+
+        public Builder() {
+
+        }
+
+        /**
+         * Sets the back-off delay used in caches.
+         */
+        public Builder withBackOffDelay(Duration delay) {
+            this.backOffDelay = Preconditions.checkNotNull(delay, "Delay cannot be null");
+            return this;
+        }
+
+        /**
+         * Sets the minimum time between two requests for caches.
+         */
+        public Builder withMinDelayBetweenRequests(Duration delay) {
+            this.minDelayBetweenRequests = Preconditions.checkNotNull(delay, "Delay cannot be null");
+            return this;
+        }
+
+        /**
+         * Enable/Disable the automatic adjustment of read timeout
+         */
+        public Builder withTimeoutAutoAdjustmentEnabled(boolean enabled) {
+            this.timeoutAutoAdjustmentEnabled = enabled;
+            return this;
+        }
+
+        /**
+         * Sets the margin of the read timeout for caches.
+         * The margin represents the additional amount of time given to the read timeout, in addition to the wait duration.
+         */
+        public Builder withTimeoutAutoAdjustmentMargin(Duration margin) {
+            this.timeoutAutoAdjustmentMargin = Preconditions.checkNotNull(margin, "Margin cannot be null");
+            return this;
+        }
+
+        public CacheConfig build() {
+            return new CacheConfig(
+                    backOffDelay != null ? backOffDelay : DEFAULT_BACKOFF_DELAY,
+                    minDelayBetweenRequests != null ? minDelayBetweenRequests : DEFAULT_MIN_DELAY_BETWEEN_REQUESTS,
+                    timeoutAutoAdjustmentEnabled != null ? timeoutAutoAdjustmentEnabled : DEFAULT_TIMEOUT_AUTO_ADJUSTMENT_ENABLED,
+                    timeoutAutoAdjustmentMargin != null ? timeoutAutoAdjustmentMargin : DEFAULT_TIMEOUT_AUTO_ADJUSTMENT_MARGIN);
+        }
```

Comment:
*Opinionated comment*

I would personally prefer to initialize the builder with the default values. 
* It makes the `build()` method simpler (no ternary operators everywhere)
* I find the declaration of those default values easier to read

Reply:
done.